### DISCUSSION TITLE
I want to use ECG_ An error occurred while Peak was labeling the R-point for my own ECG signal

### DISCUSSION THREAD

**Discussion by lili084** [neuropsychology/NeuroKit#930] [Q&A] [answered]
I have an ECG signal with a sampling frequency of 200Hz and a length of 5280000. After modifying the corresponding sampling frequency and variable name, an error was reported.
`from scipy.io import loadmat
# Load NeuroKit and other useful packages
import neurokit2 as nk
import numpy as np
import pandas as pd
# 读取.mat文件
ecg1 = loadmat('D:/signal/1.mat')
ecg= ecg1['signal']
ecg.T
rpeaks, info = nk.ecg_peaks(ecg, sampling_rate=200)`
The error prompt is【ValueError: Shape of array too small to calculate a numerical gradient, at least (edge_order + 1) elements are required.】
I don't know how to solve it, but I still report this error when intercepting a certain segment of ECG signal input. I look forward to your reply very much.

**Comment by DominiqueMakowski** [comment]
seems like it's a scipy error rather than neurokit

**Reply by lili084** [reply]
I was able to run the example and produce results normally, so I think my environment configuration should be fine, can you please elaborate on what could be the reason?

**Reply by lili084** [reply, marked as answer]
I have resolved the issue. The error stemmed from the fact that my data is a two-dimensional array, whereas the input ECG data should actually be a one-dimensional array. Using "reshape" to convert it to a one-dimensional array solved the problem.
